test/new-e2e/pkg/environments/aws/docker/host.go

```diff
@@ -120,20 +122,26 @@ func WithoutAgent() ProvisionerOption {
 	}
 }
 
+// WithAwsEnvironment sets the AWS environment to use
+// If not set, a new environment will be created
+func WithAwsEnvironment(env *aws.Environment) ProvisionerOption {
```

Comment:
nit: Given that we are in an AWS provisioner already (and packages are `aws*`), `func WithEnvironment(env *aws.Environment)` would be enough I think.